src/js/components/stock-movement/inbound/StockMovementInboundFilters.jsx

```diff
@@ -0,0 +1,209 @@
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Origin',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchLocations,
+    }) => ({
+      loadOptions: fetchLocations,
+    }),
+  },
+  destination: {
+    type: FilterSelectField,
+    attributes: {
+      valueKey: 'id',
+      filterElement: true,
+      placeholder: 'Destination',
+      showLabelTooltip: true,
+      disabled: true,
+    },
+  },
+  requestedBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Requested By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  createdBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Created By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  updatedBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Updated By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  createdAfter: {
+    type: DateFilter,
+    attributes: {
+      label: 'react.stockMovement.filter.createdAfter.label',
+      defaultMessage: 'Created after',
+      dateFormat: 'MM/DD/YYYY',
+      filterElement: true,
+    },
+  },
+  createdBefore: {
+    type: DateFilter,
+    attributes: {
+      label: 'react.stockMovement.filter.createdBefore.label',
+      defaultMessage: 'Created before',
+      dateFormat: 'MM/DD/YYYY',
+      filterElement: true,
+    },
+  },
+};
+
+const StockMovementInboundFilters = ({
+  setFilterParams,
+  debounceTime,
+  minSearchLength,
+  filterParams,
+  fetchStatuses,
+  shipmentStatuses,
+  isShipmentStatusesFetched,
+}) => {
+  const fetchUsers = debounceUsersFetch(debounceTime, minSearchLength);
+  const fetchLocations = debounceLocationsFetch(debounceTime, minSearchLength, [], true);
+
+
+  useEffect(() => {
+    if (!isShipmentStatusesFetched) fetchStatuses();
```

Comment:
I would add `!isShipmentStatusesFetched || shipmentStatuses.length === 0` in case somehow it fetched empty list and we would want to refetch it (low possibiltiy to happen)